src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,545 @@
+
+function gpkgread(fname; kwargs... )
+    # Requirement 3: File name has to end in ".gpkg"
+    if !(endswith(fname,".gpkg")) @error "the file extension is not .gpkg'" end
```

Comment:
We don't need to check the file extension here. This `gpkgread` function is internal and only used by `GeoIO.load`, which already checks the extension.